distsql/distsql.go

```diff
@@ -131,6 +136,44 @@ func (r *selectResult) NextRaw() ([]byte, error) {
 	return re.result, errors.Trace(re.err)
 }
 
+// Read reads data to the chunk.
+func (r *selectResult) Read(chk *chunk.Chunk) error {
+	chk.Reset()
+	re := <-r.results
+	if re.err != nil {
+		return errors.Trace(re.err)
+	}
+	if re.result == nil {
+		return nil
+	}
+	// As we will use streaming soon, we can just write all data to a single *chunk.Chunk.
+	resp := new(tipb.SelectResponse)
+	err := resp.Unmarshal(re.result)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for i := range resp.Chunks {
+		err = r.readRowsData(chk, resp.Chunks[i].RowsData)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+	return nil
+}
+
+func (r *selectResult) readRowsData(chk *chunk.Chunk, rowsData []byte) error {
+	var err error
+	for len(rowsData) > 0 {
+		for i := 0; i < r.rowLen; i++ {
+			rowsData, err = codec.DecodeOneToChunk(rowsData, chk, i, r.fieldTypes[i], r.loc)
```

Comment:
We should implement some mechanism to make sure we don't exceed the max capacity of `chk` to control the memory usage.

Reply:
That's another thing and irrelevant to this PR. @zz-jason